flink-cdc-runtime/src/test/java/org/apache/flink/cdc/runtime/operators/transform/UnifiedTransformOperatorTest.java

```diff
@@ -0,0 +1,919 @@
+                .deletePostTransformed("id001", "ID001", 18, "reference001")
+                .runTests()
+                .destroyHarness();
+    }
+
+    @Test
+    public void testWildcardTransform() throws Exception {
+        TableId tableId = TableId.tableId("my_company", "my_branch", "wildcard");
+        UnifiedTransformTestCase.of(
+                        tableId,
+                        "*, id + age as computed",
+                        "id > 100",
+                        Schema.newBuilder()
+                                .physicalColumn("id", DataTypes.INT())
+                                .physicalColumn("name", DataTypes.STRING())
+                                .physicalColumn("age", DataTypes.INT())
+                                .primaryKey("id")
+                                .build(),
+                        Schema.newBuilder()
+                                .physicalColumn("id", DataTypes.INT())
+                                .physicalColumn("name", DataTypes.STRING())
+                                .physicalColumn("age", DataTypes.INT())
+                                .primaryKey("id")
+                                .build(),
+                        Schema.newBuilder()
+                                .physicalColumn("id", DataTypes.INT())
+                                .physicalColumn("name", DataTypes.STRING())
+                                .physicalColumn("age", DataTypes.INT())
+                                .physicalColumn("computed", DataTypes.INT())
+                                .primaryKey("id")
+                                .build())
+                .initializeHarness()
+                .insertSource(1000, "Alice", 17)
+                .insertPreTransformed(1000, "Alice", 17)
+                .insertPostTransformed(1000, "Alice", 17, 1017)
+                .insertSource(2000, "Bob", 18)
+                .insertPreTransformed(2000, "Bob", 18)
+                .insertPostTransformed(2000, "Bob", 18, 2018)
+                .updateSource(new Object[] {2000, "Bob", 18}, new Object[] {2000, "Barcarolle", 16})
+                .updatePreTransformed(
+                        new Object[] {2000, "Bob", 18}, new Object[] {2000, "Barcarolle", 16})
+                .updatePostTransformed(
+                        new Object[] {2000, "Bob", 18, 2018},
+                        new Object[] {2000, "Barcarolle", 16, 2016})
+                // filtered out data row
+                .insertSource(50, "Carol", 19)
+                .insertPreTransformed(50, "Carol", 19)
+                .insertPostTransformed()
+                .deleteSource(1000, "Alice", 17)
+                .deletePreTransformed(1000, "Alice", 17)
+                .deletePostTransformed(1000, "Alice", 17, 1017)
+                .runTests()
+                .destroyHarness();
+
+        UnifiedTransformTestCase.of(
+                        tableId,
+                        "id + age as computed, *",
```

Comment:
Could you add  test to cover cast case like `id + age as computed, CAST(computed AS BIGINT）as larger_computed`?

Reply:
This implementation does not support writing an expression that references another calculated column, which could be easily circumvented by expanding it to something like `CAST(id + age AS BIGINT)`.

Could we address it in another PR considering it revolves many corner cases? Like which column to use when overlapped, and how to behave if one writes a recursive expression like `A as B, B as A`.